### PR TITLE
feat: invite users to squad modal

### PR DESCRIPTION
**Pull request by el**
## Changes

SCR-20230126-lmi

### Describe what this PR does
- Short and concise, bullet points can help
- Screenshots if applicable can also help

## Events

Did you introduce any new tracking events?
Don't forget to update the [Analytics Taxonomy sheet](https://docs.google.com/spreadsheets/d/18Lv7zXges9QfVX5VYL1a-Hyl0e1sQ3sLr0OK8YZWKXI/edit#gid=0)

Log the new/changed events below:

| Type   | event_name  | value |
|--------|-------------|-------|
| Change/New | event name  | extra: { ... } |

### On those affected packages:
- [ ] Have you done sanity checks in the webapp?
- [ ] Have you done sanity checks in the extension?
- [ ] Does this not break anything in companion?

### Did you test the modified components media queries?
- [ ] MobileL (420px)
- [ ] Tablet (656px)
- [ ] Laptop (1020px)

#### Did you test on actual mobile devices?
- [ ] iOS (Chrome and Safari)
- [ ] Android

WT-{number} #done

### CODE REVIEW

File: packages/shared/src/components/squads/InviteTextField.tsx [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+const InviteText: ForwardRefRenderFunction<
+  InviteTextFieldHandle,
+  InviteTextFieldProps
+> = ({ squad, isLoading }, ref): ReactElement => {
+  const token = squad?.currentMember?.referralToken ?? '';
+  const invitation = `${squad?.permalink}/${token}`;
+  const [copying, copyLink] = useCopyLink(() => invitation);
+  useImperativeHandle(ref, () => ({
+    copyLink,
```
Comment (el) [inline packages/shared/src/components/squads/InviteTextField.tsx:30]: @sshanzel @rebelchris I have exposed the function to outside here.

**Review by rebelchris — APPROVED** [review]
Big cleanup though! Lovely to see.
Just awaiting Sab's sign off, and one small question about the hook used.

File: packages/shared/src/components/modals/common.tsx [hunk 2]
```diff
@@ -31,6 +34,7 @@ export const modals = {
   [LazyModal.EditSquad]: EditSquadModal,
   [LazyModal.PostToSquad]: PostToSquadModal,
   [LazyModal.LockedSquad]: LockedSquadModal,
+  [LazyModal.SquadInvite]: SquadInviteModal,
```
[the remark this answers is not in the document]
Comment (el) [reply packages/shared/src/components/modals/common.tsx:37]: This is becoming more and more useful, great work ❤️

File: packages/shared/src/components/modals/common/types.ts [hunk 3]
```diff
@@ -21,11 +21,12 @@ export enum ModalSize {
...
 
 export enum LazyModal {
-  NewSquad = 'newSquad',
   BetaSquad = 'betaSquad',
   EditSquad = 'editSquad',
-  PostToSquad = 'postToSquad',
   LockedSquad = 'lockedSquad',
+  NewSquad = 'newSquad',
+  PostToSquad = 'postToSquad',
+  SquadInvite = 'squadInvite',
```
[the remark this answers is not in the document]
Comment (el) [reply packages/shared/src/components/modals/common/types.ts:29]: Yeah, sort everything 😜

File: packages/shared/src/components/squads/InviteTextField.tsx (hunk 1 above)
Comment (el) [reply packages/shared/src/components/squads/InviteTextField.tsx:30]: Yeah, it shouldnt have any negative effects. Pretty useful stuff if you ask me ^_^

**Pull request merged**